poulpe-model/src/test/java/org/jtalks/poulpe/model/entity/PoulpeUserTest.java

```diff
@@ -31,4 +41,39 @@ public Group groupWithId(long id) {
         return group;
     }
 
+    @Test
+    public void theEntityFieldsShouldBeSerialized() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+        PoulpeUser user = new PoulpeUser("username","email","password","salt");
+        user.setId(1);
+        user.setAvatar(new byte[]{1});
+        user.setVersion(1L);
+        user.setFirstName("First Name");
+        user.setLastName("Last Name");
+        user.setBanReason("Ban Reason");
+        user.setEnabled(true);
+        user.setRole("Role");
+        Method setLastLogin = User.class.getDeclaredMethod("setLastLogin", DateTime.class);
+        Method setEncodedUsername = User.class.getDeclaredMethod("setEncodedUsername", String.class);
+        setLastLogin.setAccessible(true);
+        setEncodedUsername.setAccessible(true);
+        setLastLogin.invoke(user,DateTime.now());
+        setEncodedUsername.invoke(user,"Encoded Username");
+
+        byte[] serialize = SerializationUtils.serialize(user);
+        PoulpeUser serializedUser = (PoulpeUser)SerializationUtils.deserialize(serialize);
+        assertReflectionEquals(user, serializedUser);
+    }
+
+    @Test
+    public void groupsFieldIsNotSerializable(){
+        List<Group> groups = PoulpeGroup.createGroupsWithNames("The Group");
+        List<Group> serializedGroups = new ArrayList<Group>();
+        try {
+            byte[] serialize = SerializationUtils.serialize(groups);
+            serializedGroups = (List<Group>) SerializationUtils.deserialize(serialize);
+        } catch (Exception e){
```

Comment:
Okay, I have different comments then:
- I wanted to test the serialization of PoulpeUser *with* groups being set
- After deserialization we need to check that the field is `null` or an empty list

And right your test is not valid because:
- You compare original list with `null`
- You use `assertSame()` which compares the references instead of object contents